File: ios/SmileIDModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(SmileIDModule, NSObject)

RCT_EXTERN_METHOD(initialize:(NSDictionary *)config)

RCT_EXTERN_METHOD(getVersion:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

@end

File: ios/SmileIDModule.swift
import Foundation
import React

@objc(SmileIDModule)
class SmileIDModule: NSObject, RCTBridgeModule {

    static func moduleName() -> String! {
        return "SmileID"
    }

    @objc
    static func requiresMainQueueSetup() -> Bool {
        return false
    }

    @objc
    func initialize(_ config: NSDictionary) {
        // Initialize SmileID SDK with config
    }

    @objc
    func getVersion(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
        resolve("1.0.0")
    }
}
